add TensorDataSource

arnej27959 opened the pull request:
@andreer and @johansolbakken please review

arnej27959 committed:
add TensorDataSource

andreer commented on vespajlib/src/main/java/com/yahoo/tensor/TensorDataSource.java:
```diff
@@ -0,0 +1,288 @@
+            case DOUBLE:
+            case FLOAT:
+            case BFLOAT16:
+                sink.doubleValue(value);
```
Could we call sink.floatValue for float/bfloat16 and sink.byteValue for int8?

arnej27959 replied:
will do

arnej27959 committed:
fix startField condition

arnej27959 committed:
add unit test

arnej27959 committed:
Better test coverage for TensorDataSource

Expands the TensorDataSource test suite with systematic coverage
of all tensor type combinations and encoding options.

Test coverage now includes:
- Scalar tensors (0D)
- Sparse tensors (1D and 2D mapped)
- Dense tensors (1D and 2D indexed)
- Mixed tensors (all 4 combinations of mapped+indexed)
- All encoding options (short/long form, direct values, hex)
- Different value types (double, float, int8, bfloat16)

arnej27959 committed:
fix endObject bug

arnej27959 committed:
Merge branch 'master' into arnej/add-tensor-data-source

arnej27959 commented on vespajlib/src/main/java/com/yahoo/tensor/TensorDataSource.java:
```diff
@@ -0,0 +1,288 @@
+                } catch (RuntimeException e) {
+                    throw new RuntimeException(e);
+                }
+            });
```
no, will drop it

arnej27959 commented on vespajlib/src/main/java/com/yahoo/tensor/TensorDataSource.java:
```diff
@@ -0,0 +1,288 @@
+
+    private void emitSingleDimensionCells(MappedTensor tensor, DataSink sink)  {
+        if (tensor.type().dimensions().size() > 1)
+            throw new IllegalStateException("Single dimension encoding requires exactly one dimension");
```
will check != 1

arnej27959 committed:
minor changes after review feedback
